avidemux_core/ADM_coreImage/src/ADM_image.cpp

```diff
@@ -152,15 +152,16 @@ bool BitBlitAlpha(uint8_t *dst, uint32_t pitchDst,uint8_t *src,uint32_t pitchSrc
     if(alpha>255) alpha=255;
     if(alpha==255) // take a shortcut
         return BitBlit(dst,pitchDst,src,pitchSrc,width,height);
+    float af = alpha * (256.0/255.0);
```

Comment:
I'm not sure that this won't collapse into `float af = alpha * 1;` with clang. The latter may require explicit cast to float for the first operand.

Reply:
i hope not, otherwise clang would be an unusable compiler :-)

Reply:
it would violate basic Implicit type conversion rules

Reply:
Implicit conversion in the above statement verified working with clang :-)

Reply:
Interestingly, now I can revert https://github.com/mean00/avidemux2/commit/a29b29b35ee9c0b0a4be7383b7a1a05414eda400 and fadeToBlack still works fine on Monterey. I cannot rule out that Apple clang was indeed broken a while ago, but has been fixed since then. I just cannot verify that this extends to the version available for Catalina as well.